server.py

```diff
@@ -44,13 +34,24 @@ def handle_client_output(socket_to_write):
 
     if message_queues[socket_to_write] == 0:
         return
-
     try:
         socket_to_write.send(message_queues[socket_to_write])
         message_queues[socket_to_write] = 0
     except socket.error, e:
         print "Client aborted " + str(e)
 
+def handle_clients_removal(socket_to_remove):
+
+    message_queues[socket_to_remove] = 0
```

Comment:
`del message_queues[socket_to_remove]`
WIll remove this key from the dictionary, so you don't save unused sockets :)